### PR TITLE
fix(ui5-menu): focus first element after adding new items

### LINKED ISSUE #7237: [ui5-menu | Accessibility]: Focus not shifting to the first element when the menu has dynamic content
### Bug Description

When the ui5-menu menu or the sub-menu has menu items that are loaded dynamically after showing the busy indicator, the focus does not shift to the first element after the menu loads.

Video: https://sap-my.sharepoint.com/:v:/p/vignesh_r02/ER0q_uLNU6BAi_YC6vqG1XIBxyXKiXFaktR7ya0Dj_-ggw?e=tAbArb

### Affected Component

ui5-menu, ui5-menu-item

### Expected Behaviour

When i hit enter on a menu or sub-menu with dynamic menu items, after the busy indicator, the focus should go to the first element.

### Isolated Example

https://sap-my.sharepoint.com/:v:/p/vignesh_r02/ER0q_uLNU6BAi_YC6vqG1XIBxyXKiXFaktR7ya0Dj_-ggw?e=tAbArb

### Steps to Reproduce

This can be reproduced in the ui5-documentation itself.
1. Hit tab and go to the menu with dynamic content
2. Hit enter
3. After the menu loads, the you can see the focus is not there for the first element
...

### Log Output, Stack Trace or Screenshots

_No response_

### Priority

None

### UI5 Web Components Version

1.14.0

### Browser

Chrome

### Operating System

_No response_

### Additional Context

_No response_

### Organization

SAP Labs

### Declaration

- [X] I’m not disclosing any internal or sensitive information.

### PR DESCRIPTION
**Pull request by tsanislavgatev**
to do decription. WIP.

fixes: https://github.com/SAP/ui5-webcomponents/issues/7237

### CODE REVIEW

**Review by unazko — CHANGES_REQUESTED** [review]
The solution is working for the main menu. We'll need to fix it for the sub-menus as well.

**Comment by tsanislavgatev** [comment]
> The solution is working for the main menu. We'll need to fix it for the sub-menus as well.

That was left out during the initial discussion. But the same code was added to the menu item, so the sub menus are also handled.

File: packages/main/src/Menu.ts [hunk 1]
```diff
@@ -277,6 +283,25 @@ class Menu extends UI5Element {
 		this._menuItems.forEach(item => {
 			item._siblingsWithIcon = siblingsWithIcon;
 		});
+
+		if (this._newItemsAdded) {
+			if (this._menuItems.length > 0) {
+				const hasSelectedOrFocusedItem = this._menuItems.some(item => item.selected);
```
Comment (unazko) [inline packages/main/src/Menu.ts:289]: We could find the item, which will receive the focus directly
`const item = this._menuItems.find(item => item.selected) || this._menuItems[0];`
and focus it.

**Review by unazko — CHANGES_REQUESTED** [review]
The solution works.
There are lint errors that needs to be fixed though.

File: packages/main/src/Menu.ts (hunk 1 above)
Comment (tsanislavgatev) [reply packages/main/src/Menu.ts:289]: Yes, the current version was test only purposes, not a real PR. Now it's time to polish the code.

**Review by unazko — APPROVED** [review]

**Review by nnaydenow — DISMISSED** [review]
Blocking unti: https://github.com/SAP/ui5-webcomponents/issues/10070

**Comment by tsanislavgatev** [comment]
New approach implementing, dont merge.

**Comment by tsanislavgatev** [comment]
Closing due to change of approach. Will be fixed in a separate PR linked to the same issue.

**Pull request closed without merging**